src/Featurizers/Structs.h

```diff
@@ -240,6 +299,56 @@ bool SingleValueSparseVectorEncoding<T>::operator!=(SingleValueSparseVectorEncod
     return (*this == other) == false;
 }
 
+
+// ----------------------------------------------------------------------
+// |
+// |  CSREncoding
+// |
+// ----------------------------------------------------------------------
+template <typename T>
+CSREncoding<T>::CSREncoding() {
+    RowIndex.push_back(0);
+}
+
+template <typename T>
+CSREncoding<T>::CSREncoding(std::vector<value_type> values, std::vector<std::uint64_t> col, std::vector<std::uint64_t> row) :
+    Values(std::move(values)),
+    ColIndex(std::move(col)),
+    RowIndex(std::move(row)) {
+        if(RowIndex[0] != 0) {

```

Comment:
>if(RowIndex[0] != 0) { [](start = 8, length = 22)

Will this statement be true for a matrix that doesn't have any values in the first row?